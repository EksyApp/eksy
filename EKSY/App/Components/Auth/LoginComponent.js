import React, {Component} from 'react'
import {Text, View, StyleSheet, ScrollView} from "react-native";
import {Button, Header} from "../Common";
import * as Theme from "../../Theme/";
import Input from "../Common/Input";
import Message from "../Common/Message";
import {Actions} from 'react-native-router-flux'
import {CheckBox} from "../Common/CheckBox";

class LoginComponent extends Component {

	render() {
		return (
				<View style={styles.container}>
					<Header title='Login' backButton/>
					<ScrollView>
						<View>
							<Input
									label='Email'
									onChangeText={(email) => this.props.onEmailChange(email)}
									placeholder='Email'
									keyboardType="email-address"
									returnKeyType="next"
							/>
						</View>
						<View>
							<Input
									label='Password'
									onChangeText={password => this.props.onPasswordChange(password)}
									placeholder='Password'
									secureTextEntry
									autoCorrect={false}
									onSubmitEditing={() => {
										this.props.onLoginClick()
									}}
							/>
						</View>
						<View style={styles.section}>
							<Message>{this.props.response}</Message>
						</View>
						<View style={styles.section}>
							<Button onPress={() => this.props.onLoginClick()}>
								Log in
							</Button>
						</View>
						<View style={[styles.section, styles.bottomMargin]}>
							<Message>Not signed up yet?</Message>
							<Button onPress={() => {Actions.signUp()}}>Sign up</Button>
						</View>
					</ScrollView>
				</View>
		)
	}

}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor,
	},

	bottomMargin: {
		marginBottom: 30
	},

	section: {
		marginTop: 20
	},

})

export default LoginComponent
