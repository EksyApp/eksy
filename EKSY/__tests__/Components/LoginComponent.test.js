import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LoginComponent from "../../App/Components/Auth/LoginComponent";

test('LoginComponent renders correctly', () => {
	const tree = renderer.create(
			<LoginComponent
					onEmailChange = {() => {} }
					onPasswordChange = {() => {} }
					onLoginClick = {() => {} }
					response = ''
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});