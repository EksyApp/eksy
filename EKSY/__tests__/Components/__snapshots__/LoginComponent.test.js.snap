// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Login
    </Text>
  </View>
  <RCTScrollView
    backButton={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgb(243,244,229)",
          "flex": 1,
        },
        undefined,
      ]
    }
    title="Login"
  >
    <View>
      <View>
        <View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    Object {
                      "color": "rgb(101,114,122)",
                      "fontSize": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Email
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#bdc6cf",
                  "borderBottomWidth": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
                Object {
                  "backgroundColor": "rgb(226,227,217)",
                  "margin": 10,
                },
              ]
            }
          >
            <TextInput
              keyboardType="email-address"
              label="Email"
              onChangeText={[Function]}
              placeholder="Email"
              returnKeyType="next"
              selectionColor="#86939e"
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 17.5,
                    "height": 36,
                    "width": 750,
                  },
                  Object {
                    "color": "rgb(89,85,78)",
                    "fontSize": 18,
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View>
        <View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    Object {
                      "color": "rgb(101,114,122)",
                      "fontSize": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Password
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#bdc6cf",
                  "borderBottomWidth": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
                Object {
                  "backgroundColor": "rgb(226,227,217)",
                  "margin": 10,
                },
              ]
            }
          >
            <TextInput
              autoCorrect={false}
              label="Password"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              selectionColor="#86939e"
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 17.5,
                    "height": 36,
                    "width": 750,
                  },
                  Object {
                    "color": "rgb(89,85,78)",
                    "fontSize": 18,
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(89,85,78)",
                "fontSize": 18,
                "textAlign": "center",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "rgb(230,232,227)",
              "borderColor": "rgb(101,114,122)",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 15,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "rgb(101,114,122)",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            Log in
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 20,
            },
            Object {
              "marginBottom": 30,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(89,85,78)",
                "fontSize": 18,
                "textAlign": "center",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          Not signed up yet?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "rgb(230,232,227)",
              "borderColor": "rgb(101,114,122)",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 15,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "rgb(101,114,122)",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            Sign up
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
